### DISCUSSION TITLE
code-server instllation on google cloud

### DISCUSSION THREAD

**Discussion by OOassafOO** [coder/code-server#6010] [Q&A]
I'm trying to install code-server on google cloud VM, and possibly missing something. My instillation:

```
curl -fsSL https://code-server.dev/install.sh | sh
sed -i.bak 's/auth: password/auth: none/' ~/.config/code-server/config.yaml
sudo systemctl restart code-server@$USER
ssh -N -L 8080:127.0.0.1:8080 [user]@<instance-ip>
```

In the last command line should instance-ip be the EXTERNAL_IP address (for example by running: 'gcloud compute instances list')
and the user is $USER ?
for this setting I'm getting: 
`Permission denied (publickey)`

also, which address should be added in the browser ?

thanks a lot for your time,
Assaf

**Comment by code-asher** [comment]
It looks like you are forwarding 8080 so you can just go to 
localhost:8080 in the browser.

Also it looks like you are installing code-server on your local 
machine and then connecting to your VM via SSH?  I think what you 
may want to do is install code-server on the VM so you would SSH 
first and then perform the installation.  Then you can exit and do 
the SSH port forwarding.

As for the permission denied error, have you uploaded your public 
SSH key to the VM?

https://cloud.google.com/compute/docs/connect/add-ssh-keys
https://github.com/coder/code-server/blob/45aef719d32f23d65748662be3725035de723618/docs/requirements.md?plain=1#L19
